docs/streams/developer-guide/dsl-topology-naming.html

```diff
@@ -0,0 +1,290 @@
+              <a href="/{{version}}/documentation/streams/tutorial">Tutorial: Write App</a>
+            </div -->
+        </div>
+    </div>
+
+    <div class="section" id="naming">
+        <span id="streams-developer-guide-dsl-topology-naming"></span>
+        <h1>Naming Kafka Streams DSL Appliations<a class="headerlink" href="#naming" title="Permalink to this headline"></a></h1>
+
+
+		With the 2.4 release, you now can give names to processors when using the Kafka 
+		Streams DSL.  Unlike the PAPI, where you are required to give a name for the processor, in the 
+		DSL, the names are generated for you.  The name generation strategy is 
+		<code>KSTREAM|KTABLE-&gt;operator name&lt;-&gt;number suffix&lt;</code>. The number suffix is a 
+		globally incrementing number that represents the operator's order in the topology.  
+		The number prefixed with a varying number of "0"s to create a suffix that is consistently 10 
+		characters long.   While the internal naming makes creating a topology with the DSL much more straightforward, 
+		there are a couple of trade-offs.  The first trade-off is what we could consider cognitive, and the other 
+		more severe trade-off is the shifting of repartition topic, 
+		changelog topic, and state store names.
+
+
+		<h2>Cognitive Issues</h2>
+
+		By saying there is a cognitive trade-off, we are referring to viewing a description of the topology. 
+		When you render the string description of your topology via the <code>Topology#desribe()</code> 
+		method, you can see what the processor is, but you don't have any context for their business purpose.  
+		For example, consider the following simple topology:
+
+		<code>
+		   builder.&gt;String,String&lt;stream("input")
+		          .filter((k,v) -> !v.equals("invalid_txn"))
+		          .mapValues((v) -> v.substring(0,5))
+		          .to("output")
+		</code>
+
+		When running <code>Topology#describe()</code> yields this string:
+
+		<code>
+		Topologies:
+		   Sub-topology: 0
+		    Source: KSTREAM-SOURCE-0000000000 (topics: [input])
+		      --> KSTREAM-FILTER-0000000001
+		    Processor: KSTREAM-FILTER-0000000001 (stores: [])
+		      --> KSTREAM-MAPVALUES-0000000002
+		      <-- KSTREAM-SOURCE-0000000000
+		    Processor: KSTREAM-MAPVALUES-0000000002 (stores: [])
+		      --> KSTREAM-SINK-0000000003
+		      <-- KSTREAM-FILTER-0000000001
+		    Sink: KSTREAM-SINK-0000000003 (topic: output)
+		      <-- KSTREAM-MAPVALUES-0000000002
+		 </code>
+
+		 From this report, we can see what the different operators are, but what is the broader context here?  
+		 For example, consider <code>KSTREAM-FILTER-0000000001</code>, we can see it's a 
+		 filter operation meaning records are dropped that don't match the given predicate.  But what is 
+		 the meaning of the predicate?  Additionally, we can see the topic names of the source and sink nodes, 
+		 but what if the topics aren't named in a meaningful way?  Then we're left to guess the 
```

Comment:
> aren't named in a meaningful way

Are you referring to repartition topics, or users failing to name their topics meaningfully? We should definitive make it clear they are encouraged to name things meaningfully, and it seems weird to expect users to choose meaningful names for their operators but not for their topics